File: Core/Models/Entities/Product.cs
namespace Core.Models.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string ImageUrl { get; set; }
        public ProductType Type { get; set; }
        public int TypeId { get; set; }
        public ProductBrand Brand { get; set; }
        public int BrandId { get; set; }
        public int Stock { get; set; }
    }
}

File: client/src/app/models/product.ts
export interface IProduct {
    id: number;
    createdBy: string;
    createdDate: string;
    updatedBy: string;
    updatedDate: string;
    isActive: number;
    name: string;
    description: string;
    price: number;
    imageUrl: string;
    type: Type;
    typeId: number;
    brand: Brand;
    brandId: number;
    stock: number;
}

export interface Type {
    id: number;
    createdBy: string;
    createdDate: string;
    updatedBy: string;
    updatedDate: string;
    isActive: number;
    name: string;
}

export interface Brand {
    id: number;
    createdBy: string;
    createdDate: string;
    updatedBy: string;
    updatedDate: string;
    isActive: number;
    name: string;
}